### PR TITLE
VIH-7419 oidc client library

### PR DESCRIPTION
**Pull request by surfsquid**
### JIRA link (if applicable) ###

added SSO/oidc to admin web

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: AdminWebsite/AdminWebsite/ClientApp/src/app/security/auth-config.module.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import { HttpClientModule, HttpEvent, HttpHandler, HttpInterceptor, HttpRequest, HTTP_INTERCEPTORS } from '@angular/common/http';
+import { APP_INITIALIZER, Injectable, NgModule } from '@angular/core';
+import { AuthInterceptor, AuthModule, LogLevel, OidcConfigService, OidcSecurityService } from 'angular-auth-oidc-client';
+import { Observable } from 'rxjs';
+import { ConfigService } from '../services/config.service';
+
+@Injectable()
+export class RefreshTokenParameterIntercepter implements HttpInterceptor {
```
Comment (levibutler) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/security/auth-config.module.ts:8]: move to own file, also can copy tests from video web :)

File: AdminWebsite/AdminWebsite/ClientApp/src/app/security/login.component.ts [hunk 2]
```diff
@@ -12,36 +14,53 @@ import { WindowRef } from '../shared/window-ref';
...
+                            );
+                            this.router.navigate(['/']);
+                        }
+                    } else {
+                        this.logger.debug(`${this.loggerPrefix} User not authenticated. Logging in`);
+                        try {
+                            this.returnUrlService.setUrl(returnUrl);
+                            this.oidcSecurityService.authorize();
+                        } catch (err) {
+                            this.logger.error(`${this.loggerPrefix} - Authorize Failed`, err);
```
Comment (levibutler) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/security/login.component.ts:60]: - already added as part of prefix

File: AdminWebsite/AdminWebsite/ClientApp/src/app/services/app-insights-logger.service.ts [hunk 3]
```diff
@@ -1,25 +1,28 @@
...
+            this.appInsights = new ApplicationInsights({
+                config: {
+                    instrumentationKey: settings.instrumentation_key
+                }
+            });
+            this.appInsights.loadAppInsights();
+            this.oidcService.userData$.subscribe(ud => {
+                this.appInsights.addTelemetryInitializer((envelope: ITelemetryItem) => {
+                    envelope.tags['ai.cloud.role'] = 'vh-admin-web';
+                    envelope.tags['ai.user.id'] = ud.userName.toLowerCase();
```
Comment (levibutler) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/services/app-insights-logger.service.ts:23]: not called username anymore?

File: AdminWebsite/AdminWebsite/ClientApp/src/app/testing/mocks/MockOidcSecurityService.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    email?: string;
+    email_verified?: false;
+    given_name?: string;
+    role?: string;
+    amr?: string;
+}
+
+export class MockOidcSecurityService {
+    userData: UserData;
+    authenticated: boolean;
```
Comment (levibutler) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/testing/mocks/MockOidcSecurityService.ts:14]: if adding additional tests will need to add configuration in here too, can copy from video web

File: AdminWebsite/AdminWebsite/appsettings.json [hunk 5]
```diff
@@ -1,5 +1,6 @@
 {
   "AzureAd": {
+    "Authority": "https://sts.windows.net/",
```
Comment (levibutler) [inline AdminWebsite/AdminWebsite/appsettings.json:3]: this need to be reverted back to login.microsoft url

Comment (levibutler) [reply AdminWebsite/AdminWebsite/appsettings.json:3]: Done

File: AdminWebsite/AdminWebsite/ClientApp/src/app/shared/shared.module.ts [hunk 6]
```diff
@@ -12,6 +12,7 @@ import { WindowRef } from './window-ref';
 import { LongDatetimePipe } from './directives/date-time.pipe';
 import { WindowScrolling } from './window-scrolling';
 import { ClipboardModule } from 'ngx-clipboard';
+import { OidcSecurityService } from 'angular-auth-oidc-client';
```
Comment (levibutler) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/shared/shared.module.ts:15]: used?

File: AdminWebsite/AdminWebsite/ClientApp/src/app/security/login.component.ts (hunk 2 above)
Comment (levibutler) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/security/login.component.ts:60]: Done

File: AdminWebsite/AdminWebsite/ClientApp/src/app/services/app-insights-logger.service.ts (hunk 3 above)
Comment (levibutler) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/services/app-insights-logger.service.ts:23]: Changed to preferred_username

File: AdminWebsite/AdminWebsite/ClientApp/src/app/security/auth-config.module.ts (hunk 1 above)
Comment (levibutler) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/security/auth-config.module.ts:8]: fixed, also fixed typo in both here and video web

File: AdminWebsite/AdminWebsite/ClientApp/src/app/testing/mocks/MockOidcSecurityService.ts (hunk 4 above)
Comment (levibutler) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/testing/mocks/MockOidcSecurityService.ts:14]: added

File: AdminWebsite/AdminWebsite/ClientApp/src/app/shared/shared.module.ts (hunk 6 above)
Comment (levibutler) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/shared/shared.module.ts:15]: Removed

File: AdminWebsite/AdminWebsite/ClientApp/src/app/security/auth-config.module.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';
+import { APP_INITIALIZER, NgModule } from '@angular/core';
+import { AuthInterceptor, AuthModule, LogLevel, OidcConfigService, OidcSecurityService } from 'angular-auth-oidc-client';
+import { ConfigService } from '../services/config.service';
+import { RefreshTokenParameterInterceptor } from './refresh-token-parameter.interceptor';
+
+export function loadConfig(configService: ConfigService, oidcConfigService: OidcConfigService): Function {
+    return () => {
+        configService.getClientSettings().subscribe(clientSettings => {
+            console.error(clientSettings);
```
Comment (levibutler) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/security/auth-config.module.ts:10]: we can remove this error now :)

**Review by shaed-parkar — APPROVED** [review]
LGTM

**Pull request merged**